bin/cylc-cat-log

```diff
@@ -271,6 +284,32 @@ def main():
         else:
             owner, host = (None, user_at_host)
 
+    cylc_tmpdir = GLOBAL_CFG.get_tmpdir()
+    if options.geditor:
+            editor = GLOBAL_CFG.get(['editors', 'gui'])
+    elif options.editor:
+            editor = GLOBAL_CFG.get(['editors', 'terminal'])
+
+    if options.editor or options.geditor:
+        if os.path.exists(filename):
+            lines = read_and_proc(filename)
+        else:
+            print >> sys.stderr, 'No such file or directory: %s' % filename
+            sys.exit(1)
+
+        viewfile = NamedTemporaryFile(
+            suffix='.' + os.path.basename(filename),
+            prefix=suite + '.', dir=cylc_tmpdir
```

Comment:
One more problem found while testing: `prefix=suite` doesn't work for suite names containing `/` corresponding to the directory path under `cylc-run/` - this makes the viewfile path refer to a non-existent directory.  Can you change to this:
`prefix=suite.replace('/','_') + '.'` - and also do the same for the original code in `cylc view`.